tests/execute/result/subresults/test.fmf

```diff
@@ -0,0 +1,32 @@
+/pass:
+    summary: Basic test of subresults
+    test: |
+        tmt-report-result /subtest/good0 PASS
+        tmt-report-result /subtest/good1 PASS
+        tmt-report-result /subtest/good2 PASS
+
+/fail:
+    summary: Reduced outcome of subresults must be fail
+    test: |
+        tmt-report-result /subtest/good PASS
+        tmt-report-result /subtest/fail FAIL
+        tmt-report-result /subtest/weird WARN
+
+/beakerlib:
+    summary: Beakerlib rlPhaseEnd as tmt subresult
+
+    # Explicitly set the TESTID to non-empty value. Also, set the
+    # BEAKERLIB_COMMAND_REPORT_RESULT to `tmt-report-result` explicitly. The
+    # command in this variable gets called with every `rlPhaseEnd`.
+    #
+    # Refs:
+    # - https://github.com/teemtee/tmt/issues/2826#issue-2225993479
+    # - https://github.com/teemtee/tmt/issues/2826#issuecomment-2085014960
+    environment:
```

Comment:
We should discuss the handling of the `TESTID` and `BEAKERLIB_COMMAND_REPORT_RESULT` env variables. These variables must be set to generate the subresults from `rlPhaseEnd` macros. Without them, only a "parent" result is generated. The `TESTID` for example, could be maybe set to the parent result's `serial-number`.

Do you think it's a good idea to somehow set/override these variables directly in the tmt or should we just keep them unset, create documentation, and let the user decide?

For more info, please see:
- https://github.com/teemtee/tmt/issues/2826#issuecomment-2085014960
- https://github.com/beakerlib/beakerlib/blob/4321c54e2e99f74b91c9376aa4324dafd0300c3d/src/testing.sh#L1075

CC: @psss @happz @lukaszachy

Reply:
As discussed today, the env variables are set internally by tmt the same way as for `BEAKERLIB_COMMAND_SUBMIT_LOG`. I also tried adding a test that checks whether these values are actually set.